client.go

```diff
@@ -0,0 +1,179 @@
+package luasandbox
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/calyptia/api/types"
+)
+
+type Client struct {
+	url        string
+	httpClient http.Client
+	nextId     int
+}
+
+type params struct {
+	Events []types.FluentBitLogAttrs `json:"events"`
+	Filter string                    `json:"filter"`
+}
+
+type request struct {
+	JsonRpcVersion string `json:"jsonrpc"`
+	Id             int    `json:"id"`
+	Method         string `json:"method"`
+	Params         params `json:"params"`
+}
+
+type LogResult struct {
+	Log   types.FluentBitLog
+	Error string
```

Comment:
How is that the client returns and array that each item can have an error?

It should return `([]types.FluentBitLog, error)`.

Reply:
We don't want an API that allows partial failure. If the Lua code fails on a single record we will error and not allow the insertion in cloud.

So just return the first error you encounter and stop there.

Reply:
I can make this change, but it would be useful to pass on the client which records raised errors. We can display hints in the json lines, for example.

If I return an error from the call if any, can we still pass separate error information of the records that failed? If so, what is the most idiomatic way of doing it?

Reply:
You can return a custom error that satisfies the `error` interface.

```go
type IndexedError struct {
  Index uint
  Msg   string
}

func (e IndexedError) Error() string {
  return fmt.Sprintf("%d: %s", e.Index, e.Msg)
}
```

```go
for i, result := range response {
  if result.Error != "" {
    return IndexedError{Index: uint(i), Msg: result.Error}
  }
}

```